app/main/views/users.py

```diff
@@ -19,14 +19,22 @@ def auth_user():
     json_payload = json_payload["authUsers"]
     validate_user_auth_json_or_400(json_payload)
 
-    user = User.query.filter(
-        User.email_address == json_payload['emailAddress'].lower()).first()
+    user = User.get_by_email_address(json_payload['emailAddress'].lower())
 
     if user is None:
         return jsonify(authorization=False), 404
-    elif encryption.checkpw(json_payload['password'], user.password):
+    elif encryption.authenticate_user(json_payload['password'], user):
+        user.logged_in_at = datetime.utcnow()
+        user.failed_login_count = 0
+        db.session.add(user)
```

Comment:
Should we have try-catches around these database operations?  Or should we just let it fail if there is an error updating the user?